misc/pragmas.md

```diff
@@ -0,0 +1,574 @@
+completely orthogonal.
+
+* What syntax should pragmas have in invisible XML? 
+
+* What representation should pragmas have in the XML serialization of
+an ixml grammar?
+
+* Where can pragmas appear?
+
+
+## Proposal(s)
+
+Two concrete proposals have been made; they have been given arbitrary
+names for discussion.
+
+### The hash-QName proposal
+
+In this proposal, pragmas take the form of a hash mark, a QName, and a
+string. Because ixml has two string delimiters, pragmas can nest, but
+only two deep.
+
+They can occur immediately before a terminal or nonterminal symbol,
+before the mark if any.
+
+In the XML form, they are realized as extension attributes whose name
+is the QName given in the pragma and whose value is the value of the
+string.
+
+For example:
+
+````
+    #my:color 'blue' a = b, #my:flavor 'vanilla' c?.
+````
+
+The corresponding XML form is:
+
+````
+    <rule name="a" my:color="blue">
+        <alt>
+            <nonterminal name="b"/>
+            <option>
+                <nonterminal name="c" my:flavor="vanilla"/>
+            </option>
+		</alt>
+    </rule>
+````
+	
+Annotations pertaining to a particular occurrence of a symbol appear 
+immediately before that occurrence, as here the `my:flavor` pragma. 
+
+Annotations pertaining to a rule as a whole, or which provide a 
+nonterminal with default value for some property appear immediately 
+before the nonterminal on the left-hand side of a rule, as here the 
+`my:color` pragma. 
+
+Annotations pertaining to the grammar as a whole appear immediately
+before the first rule, and thus appear as attributes on the first
+`rule` element of the grammar.
+
+There are no syntactic distinctions among these cases: if the rule
+shown is the first rule in the grammar, `my:color` might be supplying
+a default property value for `a`, or annotating the rule, or
+annotating the grammar as a whole. The difference must be carried by
+the semantics of the pragma: those defining the `my:color` pragma
+should be clear about what it means and what its scope is.
+
+### The brackets-QName proposal 
+
+In this proposal, pragmas take the form of a left square bracket, an
+optional mark, a QName, the pragma's data, and a right square bracket.
+Nested pairs of square brackets are allowed, so pragmas can nest
+arbitrarily deep.
+
+Pragmas can occur immediately before a terminal or nonterminal symbol,
+before the mark if any, or in the whitespace immediately before the
+full stop of a rule.
+
+In the XML form, the serialization of the pragma depends on its
+position and the mark following the opening bracket.
+
+* Pragmas occurring before the full stop of a rule are serialized as
+extension elements following the final `alt` of the right-hand side of
+the rule.
+
+* Pragmas occurring before a symbol are serialized as attributes if 
+marked `@` (ignoring marks on any nested pragmas). 
+
+* Pragmas occurring before a symbol are not serialized at all if
+marked `-`.
```

Comment:
I'm not happy with ending up with different data in the 'vxml' representation than the 'ixml' representation of the grammar, which this seems to imply.

How about using '-' to represent pragmata serialised as processing instructions instead?

Reply:
***This is where Michael and Tom got to in reviewing this doc 2021-11-23***

Reply:
We agreed to represent these pragmas in XML as processing instructions.